src/pyupgrade/plugins/redundant_open_modes.rs

```diff
@@ -0,0 +1,131 @@
+use rustpython_ast::{Constant, Expr, ExprKind, Located, Location};
+use rustpython_parser::lexer;
+use rustpython_parser::token::Tok;
+
+use crate::ast::helpers::{self, match_name_or_attr};
+use crate::ast::types::Range;
+use crate::autofix::Fix;
+use crate::check_ast::Checker;
+use crate::checks::{Check, CheckCode, CheckKind};
+use crate::source_code_locator::SourceCodeLocator;
+
+const OPEN_FUNC_NAME: &str = "open";
+
+enum OpenMode {
+    U,
+    Ur,
+    Ub,
+    RUb,
+    R,
+    Rt,
+    Wt,
+}
+
+impl OpenMode {
+    fn from_str(input: &str) -> Option<OpenMode> {
+        match input {
+            "U" => Some(OpenMode::U),
+            "Ur" => Some(OpenMode::Ur),
+            "Ub" => Some(OpenMode::Ub),
+            "rUb" => Some(OpenMode::RUb),
+            "r" => Some(OpenMode::R),
+            "rt" => Some(OpenMode::Rt),
+            "wt" => Some(OpenMode::Wt),
+            _ => None,
+        }
+    }
+
+    fn replacement_value(&self) -> String {
```

Comment:
Maybe `-> Option<String>` and return `None` instead of the empty string?

Reply:
Yeah for sure, changed this 👍